packages/terra-i18n-plugin/src/I18nAggregatorPlugin.js

```diff
@@ -65,20 +70,52 @@ function aggregateTranslations(options) {
   supportedLocales.forEach((language) => { languageMessages[language] = {}; });
 
   // Aggregate translation messages for the directory
-  languageMessages = aggregateDirectory(languageMessages, options.baseDirectory);
-
-  // Create the aggregated-translations directory
-  mkdirp.sync(path.resolve(options.baseDirectory, 'aggregated-translations'));
-
-  // Create a file for each language for the aggregated messages
-  supportedLocales.forEach((language) => {
-    if (language in languageMessages) {
-      fs.writeFileSync(path.resolve(options.baseDirectory, 'aggregated-translations', `${language}.js`),
-        generateTranslationFile(language, languageMessages[language]));
-    } else {
-      throw new Error(`Translation file found for ${language}.json, but translations were not loaded correctly. Please check that your translated modules were installed correctly.`);
-    }
-  });
+  languageMessages = aggregateDirectory(languageMessages, options.baseDirectory, compiler);
+  return languageMessages;
+}
+
+function aggregateTranslations(options, compiler) {
+  if (options.outputFileSystem) {
```

Comment:
You should be able to DRY this code out by setting a filesystem variable to either be the passed in file system or the standard file system. You've got alot of repeated code here.